src/components/SinglePost/SinglePost.js

```diff
@@ -1,30 +1,45 @@
-import { likePost ,dislikePost} from "../../features/post/postSlice"; 
-import { useDispatch } from "react-redux";
+import { likePost , dislikePost , addToBookmark , removeFromBookmark } from "../../features/post/postSlice"; 
+import { useDispatch , useSelector} from "react-redux";
+import { StyledPost } from "../../styled.components";
 
-import { BiBookmark,BiComment } from "react-icons/bi";
+import { BiComment } from "react-icons/bi";
 import { FaRegHeart, FaHeart } from "react-icons/fa";
+import { MdBookmarkBorder , MdBookmark } from "react-icons/md";
 import "./singlePost.css"
 
-export const SinglePost = ({ _id, content,username ,likes}) => {
+
+export const SinglePost = ({post , isBookmarkedPage = false}) => {
+    const {_id, content,username ,likes} = post
+    const {bookmarkPosts} = useSelector(state => state.post)
     const {likeCount} = likes
+    const isBookMarked = bookmarkPosts.findIndex(post => post._id === _id) === -1 ? false : true
     const dispatch = useDispatch()
+
     return(
-        <div className="padding-xs singlepost__wrapper border-radius-xs">
+        <StyledPost>
             <div className="avatar avatar-text avatar-text-sm margin-xs padding-sm">
                 {username.slice(0,2).toUpperCase()}
             </div>
             <div className="content__wrapper">
                <p className="text-sm margin-xs">{content}</p>
-               <div className="btn__wrapper padding-xs">
-                 <BiComment className="comment__icon"/>
-                 {
-                     likeCount === 0 ? 
-                     <FaRegHeart className="like__icon"onClick={()=>dispatch(likePost(_id))} /> :
-                     <FaHeart className="dislike__icon" onClick={()=>dispatch(dislikePost(_id))}/>
-                 }
-                 <BiBookmark className="bookmark__icon"/>
-               </div>
+               {
+                   isBookmarkedPage ? 
+                   <MdBookmark className="bookmark__icon" onClick={()=>dispatch(removeFromBookmark(_id))} /> : 
+                <div className="btn__wrapper padding-xs">
+                   <BiComment className="comment__icon"/>
+                    {
+                      likeCount === 0 ? 
```

Comment:
Change: you must change this line as it's only valid for the first user who likes the post, what if someone makes a new account in your app, this [post will always show as already liked post, if the same person tries to dislike it, it will throw an error

Reply:
In database, there will an array of like, from where you can compare and then show the correct like/dislike btn